packages/web/src/plugins/cluster-status/plugin.tsx

```diff
@@ -38,6 +36,6 @@ export default createPlugin('dashboard/cluster-status')
     }, [apiEndpoint]);
   })
 
-  .init(tk => {
-    tk.slots.rtkServices.add(configApi);
-  });
+  .init();
+
+RtkPlugin.overlay.appendContext({configApi});
```

Comment:
we didn't have any use cases for that currently, but if we need it at some point, we can try creating an abstraction over the [injectEndpoints](https://redux-toolkit.js.org/rtk-query/usage/code-splitting) method from RTK which is the ideal way of including APIs conditionally 🤔 other options would be looking at `replaceReducer` or `rootReducer.inject` , but we have to create the store only once, otherwise we run into problems